### ISSUE TITLE
Update to a newer version of .net framework

### ISSUE THREAD

**Issue by Mediumspider** [snakefoot/snaketail-net#163]
Would it be possible to update SnakeTail to use .net framework 4.0 or above?

**A bit of background**
Currently snaketail is built for .net framework 2.0

We've have some corporate control software installed that crashes SnakeTail at startup. It doesn't crash in the debugger, and the event log just says "The process was terminated due to an unhandled exception". 

Trial and error shows that it works if updated to at least .net 4.0. 

It would mean removing support for windows 2000, and if that's a dealbreaker I can create a fork.

**Comment by snakefoot** [comment]
If you place the `Snaketail.exe.config`-file next to the SnakeTail.exe, then it should recognize that it supports all .NET runtimes.

See also: #160

**Comment by Mediumspider** [comment]
I saw that one, but that is not the issue for us. 

The problem is a bug in our new corporate access control software and not SnakeTail. 

(We've doublechecked by temporarily disabling it)

Upgrading snaketail to build with at least .net 4.0 works.

I can prepare the pull request with the necessary changes if that is of interest, but I can also just create a fork if you prefer to stick to .net 2.0

**Comment by snakefoot** [comment]
What if you upgrade to .NET 3.5 ? (Instead of .NET 4.0)

**Comment by Mediumspider** [comment]
I tried 2.0, 3.0 and 3.5 and they all crashed. 
However I did a bit more digging and succeeded in getting a crash-dump using ProcDump.exe, where I could see the actual error.

``Mixed mode assembly is built against version ‘v2.0.50727′ of the runtime and cannot be loaded in the 4.0 runtime without additional configuration information.``

My guess is that the access control software injects some native code that crashes when it tries to load the native dll. A solution for me was to add the following to add "useLegacyV2RuntimeActivationPolicy" to the config file:

```xml
<configuration>
  <startup useLegacyV2RuntimeActivationPolicy="true">
...
  </startup>
</configuration>
```

It would be nice if this could be added to the config bundled with the official installer, but I acknowledge that ours is probably not a common use-case. 

I'll inform our IT department about the workaround, and update our custom installers.

*Sources*: 
* https://stackoverflow.com/q/6425707/5686117
* https://stackoverflow.com/questions/1604663/what-does-uselegacyv2runtimeactivationpolicy-do-in-the-net-4-config
* http://web.archive.org/web/20130128072944/http://www.marklio.com/marklio/PermaLink,guid,ecc34c3c-be44-4422-86b7-900900e451f9.aspx
